[cyhni] iP

cyhni opened the pull request:
# Billy
> "Billy is billy" — Billy

Billy has dedicated his whole life to remembering what you have to do! 
### Billy is
- **billy**  :fire::fire::fire:
- able to remember all your tasks
- yours

### To get your own billy,
1. Download it from [HERE](https://github.com/cyhni/ip/releases/tag/A-Jar)
2. Copy the jar file into an empty folder
3. Open a command window in that folder
4. Run the command `java -jar "billy.jar"`

### Features
- [x] Manage various tasks (Completed)
- [ ] GUI for Billy (Pending)
- [ ] Autocomplete commands (Pending)
- [ ] ~~Sentience (Pending)~~

### Here is the logo for Billy
```python
public static final String LOGO_STRING = """
            ,---,.            ,--,    ,--,
          ,'  .'  \\  ,--,   ,--.'|  ,--.'|
        ,---.' .' |,--.'|   |  | :  |  | :
        |   |  |: ||  |,    :  : '  :  : '
        :   :  :  /`--'_    |  ' |  |  ' |        .--,
        :   |    ; ,' ,'|   '  | |  '  | |      /_ ./|
        |   :     \\'  | |   |  | :  |  | :   , ' , ' :
        |   |   . ||  | :   '  : |__'  : |__/___/ \\: |
        '   :  '; |'  : |__ |  | '.'|  | '.'|.  \\  ' |
        |   |  | ; |  | '.'|;  :    ;  :    ; \\  ;   :
        |   :   /  ;  :    ;|  ,   /|  ,   /   \\  \\  ;
        |   | ,'   |  ,   /  ---`-'  ---`-'     :  \\  \\
        `----'      ---`-'                       \\  ' ;
                                                  `--`""";
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

cyhni committed:
Updated .gitignore to ignore message.txt file.

message.txt file will be used to draft commit messages and therefore does not need to be commited as a file.

cyhni committed:
Updated chatbot name and abstracted string logo to constants folder

Chatbot name updated to Billy. Logo design abstracted to a constant file that will contain all the design related string constants.

cyhni committed:
Added Greet and Exit system messages

Private static methods for printing of the introduction and saying bye.

cyhni committed:
Added echo function

cyhni committed:
Added Billy.parseCommand(String userCmd) and Billy.list()

Billy.parseCommand(String userCmd) default action is to add to the String array containing the list.
If list is inputted, it will call Billy.list() to display the current list.

cyhni committed:
Created Task class to facilitate the functions of a task.

This class represents a task and manages the done state of the tasks.

cyhni committed:
Mark and unmark as done implemented for tasks.

Edited Billy.parseCommand() to allow for the parsing of marking and umarking of tasks.

cyhni committed:
Implemented Todo, Deadline and Event classes that inherits from Task class.

cyhni committed:
Updated Billy.parseCommand() to receive todo, deadline and event commands.

Each command will parse the request and create the relevant Task childs.

cyhni committed:
Setup test using the I/O redirection technique.

cyhni committed:
Added error handling for incorrect fields, unknown task number and unknown command.

3 handled exception classes were created to handle the above errors. BillyUnknownException, BillyFieldErrorException and BillyUnknownTaskNumException.

cyhni committed:
Added delete functionality

Converted normal array used to store tasks to a ArrayList<Task>. Easily allows dynamic adding and removing of tasks from the list.

cyhni committed:
Save the tasks to a local file billy.txt

cyhni committed:
Merge branch 'branch-Level-7'

cyhni committed:
Bug fix 'event' command when /from and /to are swapped

cyhni committed:
Add constraint check on date and time formatting for 'deadline' and 'event' commands

cyhni committed:
Merge branch 'branch-Level-8'

cyhni committed:
Update some code to comply with coding standard

cyhni committed:
Extract Ui class to manage interaction with user

cyhni committed:
Extract FileManager and Parser class

cyhni committed:
Merge branch 'branch-A-MoreOOP'

cyhni committed:
Divide classes into packages

cyhni committed:
Merge branch 'branch-A-Packages'

cyhni committed:
Merge branch 'add-gradle-support'

cyhni committed:
Move main Billy.java into package

cyhni committed:
Edit code using checkstyle checker

cyhni committed:
Extract Command and TasksList class

cyhni committed:
Add JUnit tests

cyhni committed:
Merge branch 'branch-A-JUnit'

cyhni committed:
Add JavaDoc comments to all non-private classes and methods

cyhni committed:
Update code to meet coding standard

cyhni committed:
Add Find command to find a taks by keyword

cyhni committed:
Merge branch 'branch-A-JavaDoc'

cyhni committed:
Merge branch 'branch-A-CodingStandard'

cyhni committed:
Resolve merge conflict

cyhni committed:
Fix code to according to checkstyle

cyhni committed:
Update splitCmd variable name to commandStrings for clarity

jodielzy commented on src/main/java/billy/Billy.java:
```diff
@@ -0,0 +1,69 @@
+                ui.printError(e.getMessage());
+            }
+        }
+        ui.printBye();
```
I like the method names, they are clear and straightforward.

jodielzy commented on src/main/java/billy/parser/Parser.java:
```diff
@@ -0,0 +1,179 @@
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Pattern;
+
+import billy.command.Command;
+import billy.command.DeadlineCommand;
+import billy.command.DeleteCommand;
+import billy.command.EventCommand;
+import billy.command.FindCommand;
+import billy.command.ListCommand;
+import billy.command.MarkCommand;
+import billy.command.TodoCommand;
+import billy.command.UnmarkCommand;
+import billy.exceptions.BillyException;
+import billy.exceptions.BillyFieldErrorException;
+import billy.exceptions.BillyUnknownException;
+import billy.exceptions.BillyUnkownTaskNumException;
+import billy.tasks.Deadline;
+import billy.tasks.Event;
+import billy.tasks.TasksList;
+import billy.tasks.Todo;
+import billy.ui.Ui;
```
Would it be neater to leave a line between each package?

cyhni replied:
I thought about doing that too but I think it might cause some issues with the automated checkstyle. Will have to double check again!

jodielzy commented on src/main/java/billy/parser/Parser.java:
```diff
@@ -0,0 +1,179 @@
+     * @return The LocalDateTime object corresponding to the date string.
+     * @throws DateTimeException If an error occurs during date parsing.
+     */
+    public static LocalDateTime dateParsing(String date) throws DateTimeException {
```
Would it be better to change the method name into a verb (e.g parseDate)?

cyhni replied:
Good catch! Thanks for pointing out!

jodielzy commented on src/main/java/billy/ui/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     * @param task The task to be printed.
+     * @param counter The number of tasks in the list.
+     */
+    public final void printTaskAdded(Task task, int counter) {
```
Just nitpicking: "printAddedTask" might sound more natural.

cyhni replied:
Thanks for the suggestion 👍

jodielzy reviewed (commented):
LGTM! Overall, I think your code is well structured, naming of classes and methods are clear and straightforward.

cyhni committed:
Update code based on feedback from PR review

cyhni committed:
Update gradle build

cyhni committed:
Complete step 1 of javaFX tutorial

cyhni committed:
Migrate chatbox to have JavaFX functionalities

cyhni committed:
Update various CSS and FXML

cyhni committed:
Merge branch 'branch-Level-10'

cyhni committed:
Update bye command to exit program

cyhni committed:
Add assertions to define assumptions

cyhni committed:
UiTest.java: Remove dead code

There are unused code commented out from previous testing of a removed method Ui.printIntroduction().

Removing dead code can improve the code quality and keep code clean of clutter.

Let's
* remove test printIntroduction_introductionMessage_success()

cyhni committed:
Remove dead code from Billy.java

cyhni committed:
Change comments for main method to sound active

cyhni committed:
Refactor code from parseCommand method by extract method

cyhni committed:
Merge pull request #2 from cyhni/branch-A-Assertions

Merge branch 'branch-A-Assertions'

cyhni committed:
Merge branch 'master' into branch-A-CodeQuality

cyhni committed:
Merge pull request #3 from cyhni/branch-A-CodeQuality

Merge branch 'branch-A-CodeQuality' to 'master'

cyhni committed:
Add .yml file to set up github actions

cyhni committed:
Update .gitignore

cyhni committed:
Add extension C-FriendlierSyntax

cyhni committed:
Merge branch 'branch-BCD-Extension'

cyhni committed:
Update css

cyhni committed:
Update css

cyhni committed:
Add user guide

cyhni committed:
Fix ui image extension

cyhni committed:
Rename Ui.PNG to Ui.png

cyhni committed:
Fix file paths

cyhni committed:
Merge branch 'master' of https://github.com/cyhni/ip

cyhni committed:
Remove dead code

cyhni committed:
Merge pull request #5 from cyhni/remove-dead-code

Remove dead code

cyhni committed:
Handle error for negative indexes

cyhni committed:
Fix code style

cyhni committed:
Merge pull request #6 from cyhni/fix-bug-negative-index

Handle error for negative indexes
